ffs-shared/rule/src/commonMain/kotlin/rule/RuleEval.kt

```diff
@@ -274,6 +274,63 @@ private sealed class RuleExpr<out T> {
         }
         //endregion
 
+        //region IP matching\
+        val ipv4Regex = Regex(
+            "^((?:(?:25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)\\.)" +
+                "{3}(?:25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?))(?:/([0-9]|[12][0-9]|3[0-2]))?\$"
+        )
+        val defaultIpMask = 32
+
+        fun convertIpToLong(ipString: String): Long {
+            val ipParts = ipString.split(".")
+
+            var ipNumbers: Long = 0
+            for (i in 0..3) {
+                ipNumbers += ipParts[i].toLong() shl (24 - (8 * i))
+            }
+
+            return ipNumbers
+        }
+
+        data class MatchesIpv4(
+            val range: RuleExpr<String>,
+            val value: RuleExpr<String>
+        ) : FunctionExpr<Boolean>() {
+            override fun eval(env: JsonObject): Boolean {
+                val ipRange = ipv4Regex.matchEntire(range.castEval(env))
+                val ipValue = ipv4Regex.matchEntire(value.castEval(env))
+
+                if (ipRange == null || ipRange.groups[1] == null)
```

Comment:
Interesting. Detekt doesn't catch this. Still, we always use braces unless the condition and statement fit on a single line (which we also tend to avoid except for early returns) per [Kotlin's style guide](https://kotlinlang.org/docs/coding-conventions.html#control-flow-statements).

Reply:
Thanks for pointing this out 😊

Yes, Detekt didn't catch this one when I tried to push to commit. I will review the coding conventions and update my code formatting properly